### ISSUE TITLE
Allow using zod's preprocess on query 

### ISSUE THREAD

**Issue by ShauniArima** [orval-labs/orval#1575]
## What are the steps to reproduce this issue?

1. Have a query param that is a boolean (**checked** for example)
2. Generate `Zod` schemas using `Orval`
3. Call your service with a valued query param (**checked=true** or **checked=false**)

## What happens?

The boolean is not handled correctly. As it is a string in query parameters, the coerce always returns `true` when it is valued.

## What were you expecting to happen?

I expect the boolean to be valued correctly. As it is described in Zod's documentation, the `coerce` is not sufficient to handle this case (cf. https://zod.dev/?id=coercion-for-primitives). So we can use `preprocess` to handle this case.

There is 2 solutions possible :
- handle this behavior directly in Orval
- allow using `preprocess` for `Query` when using `Zod` as it allows more flexibility.

> Note that `preprocess` seems to be applied for the **whole object**. Maybe it would be useful to add **preprocess only on a specific field**.

## What versions are you using?

```
System:
  OS: macOS 14.6
  CPU: (8) arm64 Apple M2
  Memory: 60.17 MB / 16.00 GB
  Shell: 3.6.1 - /usr/local/bin/fish
npmPackages:
  orval: 7.0.1 => 7.0.1
  zod: ^3.23.8 => 3.23.8
```

**Comment by mbaroukh** [comment]
Hi. I run into the same issue today for hono generator and int values.
orval generates this : 
```
export const listQueryParams = zod.object({
  current: zod.number().min(1).optional(),
  pageSize: zod.number().min(1).optional(),
});
```

I suppose it should be 

```
export const listQueryParams = zod.object({
  current: zod.number({ coerce: true }).min(1).optional(),
  pageSize: zod.number({ coerce: true }).min(1).optional(),
});
```

coerce should be added for all non-string values.

**Comment by mbaroukh** [comment]
> PR is welcome!

yes, trying actually. Adding tests in hono package which doesn't have any. But I have to understand the setup :).

**Comment by melloware** [comment]
Yeah it takes a little getting used to the codebase and understanding what is going on!  Appreciate the help though!

**Comment by mbaroukh** [comment]
there is a long way to go ... Just managed to reproduce in a test case for now ...
https://github.com/mbaroukh/orval/commit/a6faf49b409a916bbf710dc43e69e5a40e8e42e1
I will be able to continue only next week.

**Comment by melloware** [comment]
Ok I assigned the ticket to you for now.

**Comment by melloware** [comment]
I just looked at this again we already support zod coerce. You can coerce all types just look at the output docs

**Comment by melloware** [comment]
See: https://orval.dev/reference/configuration/output#coerce

**Comment by melloware** [comment]
Nevermind I realize you are talking about query params
